lib/ansiblelint/skip_utils.py

```diff
@@ -0,0 +1,140 @@
+        ruamel_task_blocks = ruamel_data
+        pyyaml_task_blocks = pyyaml_data
+    elif file_type == 'playbook':
+        try:
+            pyyaml_task_blocks = _get_task_blocks_from_playbook(pyyaml_data)
+            ruamel_task_blocks = _get_task_blocks_from_playbook(ruamel_data)
+        except (AttributeError, TypeError):
+            # TODO(awcrosby): running ansible-lint on any .yml file will
+            # assume it is a playbook, check needs to be added higher in the
+            # call stack, and can remove this except
+            return pyyaml_data
+    else:
+        raise RuntimeError('Unexpected file type: {}'.format(file_type))
+
+    # get tasks from blocks of tasks
+    pyyaml_tasks = _get_tasks_from_blocks(pyyaml_task_blocks)
+    ruamel_tasks = _get_tasks_from_blocks(ruamel_task_blocks)
+
+    # append skipped_rules for each task
+    for ruamel_task, pyyaml_task in zip(ruamel_tasks, pyyaml_tasks):
+        if pyyaml_task.get('name') != ruamel_task.get('name'):
+            raise RuntimeError('Error in matching skip comment to a task')
+        pyyaml_task['skipped_rules'] = _get_rule_skips_from_yaml(ruamel_task)
+
+    return pyyaml_data
+
+
+def _get_task_blocks_from_playbook(playbook):
+    """Return parts of playbook that contains tasks, and nested tasks.
+
+    :param playbook: playbook yaml from yaml parser.
+    :returns: list of task dictionaries.
+    """
+    PLAYBOOK_TASK_KEYWORDS = [
+        'tasks',
+        'pre_tasks',
+        'post_tasks',
+        'handlers',
+    ]
+
+    task_blocks = []
+    for play, key in product(playbook, PLAYBOOK_TASK_KEYWORDS):
+        task_blocks.extend(play.get(key, []))
+    return task_blocks
+
+
+def _get_tasks_from_blocks(task_blocks):
+    """Get list of tasks from list made of tasks and nested tasks."""
+    NESTED_TASK_KEYS = [
+        'block',
+        'always',
+        'rescue',
+    ]
+
+    def get_nested_tasks(task):
+        return (
+            subtask
+            for k in NESTED_TASK_KEYS if k in task
+            for subtask in task[k]
+        )
+
+    for task in task_blocks:
+        for sub_task in get_nested_tasks(task):
+            yield sub_task
+        yield task
+
+
+def _get_rule_skips_from_yaml(yaml_input):
+    """Travese yaml for comments with rule skips and return list of rules."""
+    def traverse_yaml(obj):
+        yaml_comment_obj_strs.append(str(obj.ca.items))
+        if isinstance(obj, dict):
+            for key, val in obj.items():
+                if isinstance(val, (dict, list)):
+                    traverse_yaml(val)
+        elif isinstance(obj, list):
+            for e in obj:
+                if isinstance(e, (dict, list)):
+                    traverse_yaml(e)
+        else:
+            return
+
+    yaml_comment_obj_strs = []
+    traverse_yaml(yaml_input)
+
+    rule_id_list = []
+    for comment_obj_str in yaml_comment_obj_strs:
+        for line in comment_obj_str.split('\\n'):
```

Comment:
Are you sure this is what you want? It may be easier to understand with raw-literal in place:
```suggestion
        for line in comment_obj_str.split(r'\n'):
```